test/SlowTests/Issues/RavenDB_3417.cs

```diff
@@ -27,16 +27,25 @@ public void GetIndexingPerformanceStatisticsShouldWork()
 
                 WaitForIndexing(store);
 
-                var indexingPerformanceStatistics = store.Maintenance.Send(new GetIndexPerformanceStatisticsOperation());
+                Assert.True(WaitForValue(() =>
+                {
+                    var indexingPerformanceStatistics = store.Maintenance.Send(new GetIndexPerformanceStatisticsOperation());
 
-                Assert.Equal(7, indexingPerformanceStatistics.Length);
+                    if (indexingPerformanceStatistics.Length != 7)
+                        return false;
 
-                foreach (var stats in indexingPerformanceStatistics)
-                {
-                    Assert.NotNull(stats.Name);
-                    Assert.NotNull(stats.Performance);
-                    Assert.True(stats.Performance.Length > 0);
-                }
+                    foreach (var stats in indexingPerformanceStatistics)
+                    {
+                        if (stats.Name == null || stats.Performance == null)
+                            return false;
+
+                        if (stats.Performance.Length == 0)
+                            return false;
+                    }
+
+                    return true;
+                    
+                }, true, 5000));
```

Comment:
Previously we had 3 separate assertions. Now we aren't able to distinguish between them. We'll just get `Assert.True()` failure

Reply:
suggestion?

Reply:
correct